common/flatpak-dir.c

```diff
@@ -497,6 +497,35 @@ flatpak_remote_state_ensure_subsummary (FlatpakRemoteState *self,
   return TRUE;
 }
 
+gboolean
+flatpak_remote_state_ensure_subsummary_all_arches (FlatpakRemoteState *self,
+                                                   FlatpakDir         *dir,
+                                                   gboolean            only_cached,
+                                                   GCancellable       *cancellable,
+                                                   GError            **error)
+{
+  if (self->index_ht == NULL)
+    return TRUE; /* No subsummaries, got all arches anyway */
+
+  GLNX_HASH_TABLE_FOREACH (self->index_ht, const char *, arch)
+    {
+      g_autoptr(GError) local_error = NULL;
+
+      if (!flatpak_remote_state_ensure_subsummary (self, dir, arch, only_cached, cancellable, &local_error))
```

Comment:
```suggestion
      if (flatpak_remote_state_ensure_subsummary (self, dir, arch, only_cached, cancellable, &local_error))
```

Reply:
Hmm, this seems all weird. We shouldn't ever return TRUE here, we should continue to the next arch in the hash... I'll fix this.

Reply:
Better now?

Reply:
Looks correct but it's almost identical to the implementation of `ensure_remote_state_all_arches()`. Do we want both?

Reply:
They are not identical though, as `--cached` means "prefer cached before updating", whereas "only_cached" means fail if not cached. We could make it use the common one as a helper though. Lemme look at that.